src/server/saved_objects/service/lib/repository_provider.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectsRepository } from './repository';
+
+/**
+ * Provider for the Saved Object Reppository.
+ */
+export class SavedObjectsRepositoryProvider {
+
+  constructor({
+    index,
+    mappings,
+    onBeforeWrite
+  }) {
+    this._index = index;
+    this._mappings = mappings;
+    this._onBeforeWrite = onBeforeWrite;
+  }
+
+  getRepository(settings = {}) {
```

Comment:
Do we have any current usages where we'd want to pass the `index`/`mappings`/`onBeforeWrite` in the settings to the `getRepository` function? It feels like we might be adding something we don't need.

Reply:
Nope, no current usage. I thought this would be a little more extensible for other use cases, without adding a ton of complexity, but I can restrict this to only accept a `callCluster` function if you think that's the better approach to take

Reply:
I figure if they want to not use what we've already provided, they can do so via using the SavedObjectRepository constructor itself.